File: BackEnd/src/domain/interfaces/usecases/user/IChangePasswordUseCase.ts
/**
 * @file IChangePasswordUseCase.ts
 * @description Interfaz del caso de uso para cambiar la contraseña del usuario.
 * @module User
 * @layer Domain
 */
export interface IChangePasswordUseCase {
  execute(userId: string, currentPassword: string, newPassword: string): Promise<void>;
}

File: FrontEnd/lib/domain/interfaces/usecases/user/i_change_password_use_case.dart
/// @file i_change_password_use_case.dart
/// @description Interfaz: Cambia la contraseña del usuario.
/// @module User
/// @layer Domain
library;
abstract interface class IChangePasswordUseCase {
  /// Cambia la contraseña del usuario.
  Future<void> execute(String currentPassword, String newPassword);
}
